### ISSUE TITLE
packageRules.matchBranches

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#21286]
### What would you like Renovate to be able to do?

Support a new `packageRules` matcher called `matchBranches`

### If you have any ideas on how this should be implemented, please tell us here.

It should be an array of strings.

If a string starts with `/` then it should be treated as a regex. It would be good if we can support the `/..../i` modifier rather than just `/..../` like we have in other fields today).

If not a regex pattern then it should be treated as an exact match.

e.g. 

```
"matchBranches": ["foo", "/bar/"]
```

Naturally, this matcher could only occur post-lookup after branch names have been decided. It cannot be used to modify anything which is used _prior_ to that either (e.g. `separateMajorMinor`).

### Is this a feature you are interested in implementing yourself?

No

**Comment by RahulGautamSingh** [comment]
How will it different from the `matchBaseBranches` package rule?

**Comment by RahulGautamSingh** [comment]
Do we want to allow the `i` flag for all the other fields too or only this one?

**Comment by rarkins** [comment]
Yes but I think that should be a separate feature PR on its own before or after this one

**Comment by RahulGautamSingh** [comment]
Before 👍

**Comment by rarkins** [comment]
Update: use anyMatchRegexOrMinimatch()

**Comment by RahulGautamSingh** [comment]
Bit late to ask I guess, but what kind of use-cases does this option handle?

**Comment by rarkins** [comment]
Makes to easier to apply rules to every update in a branch instead of repeating the match package patterns again. For example one of the monorepo branches or group all branches

**Comment by RahulGautamSingh** [comment]
> It cannot be used to modify anything which is used prior to that either (e.g. separateMajorMinor).

Is this restriction on user behaviour or something we should make sure internally?

**Comment by rarkins** [comment]
We ideally should validate this, e.g. cannot modify separateMajorMinor type ones, or registryUrls, etc. We can perhaps add to this.
